nodes/persistence-node/src/services/PersistenceNodeApi.ts

```diff
@@ -0,0 +1,72 @@
+import { ethers } from "ethers";
+import express, { NextFunction, Request, Response } from "express";
+import http from "http";
+import * as IPFS from 'ipfs-core';
+import { InternalApiConfig } from "../config/InternalApiConfig";
+import { IpfsConfig } from "../config/IpfsConfig";
+import { Storage } from "../types/Storage";
+import { CacheRunner } from "./CacheRunner";
+import { Logger } from "./Logger";
+
+interface IDependencies {
+  ethersProvider: ethers.providers.Provider;
+  ensPublicResolver: ethers.Contract;
+  storage: Storage;
+  cacheRunner: CacheRunner;
+  ipfsNode: IPFS.IPFS;
+  ipfsConfig: IpfsConfig;
+  internalApiConfig: InternalApiConfig;
+  logger: Logger;
+}
+
+export class PersistenceNodeApi {
+  deps: IDependencies;
+
+  constructor(deps: IDependencies) {
+    this.deps = deps;
+  }
+
+  async run() {
+    const app = express();
+
+    app.get('/api/v0/past', handleError(async (req, res) => {
+      const blocks = req.query.blocks;
+
+      if (!blocks) {
+        res.status(422).send("Blocks parameter missing.");
+        return;
+      }
+
+      var head = {
+        'Content-Type': 'text/html; charset=UTF-8',
+        'transfer-encoding': 'chunked'
+      }
+
+      res.writeHead(206, head)
+
+      this.deps.logger.registerListener(log => {
+        console.log('LISTENER TRIGERED')
+        res.write(log)
```

Comment:
When CLI interacts with API, we want to stream our console logs there as well. Which means that we have to do something like this, inject a custom listener to logger, which will stream (`res.write()`) the contents as a chunked response to CLI.

Reply:
I get the idea, but I'm not sure we want long-running jobs with the CLI
We should probably just return a 200 response and then have separate commands to get the details of that job. We can handle that in another issue, for now, just start the processing of past blocks and return a 200 status